src/content/docs/recipes/apps/spawn-vim.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Spawn External Editor (Vim)
+sidebar:
+  order: 9
+  label: Spawn External Editor (Vim)
+---
+
+In this recipe, we will explore how to spawn an external editor (Vim) from within the TUI app. This
+example demonstrates how to temporarily exit the TUI, run an external command, and then return back
+to our TUI app.
+
+Full code:
+
+```rust collapsed title="main.rs (click to expand)"
+{{ #include @code/recipes/how-to-spawn-vim/src/main.rs }}
+```
+
+## Setup
+
+First, let's look at the main function and the event handling logic:
+
+```rust title="main.rs"
+{{ #include @code/recipes/how-to-spawn-vim/src/main.rs:action_enum }}
+
+{{ #include @code/recipes/how-to-spawn-vim/src/main.rs:main }}
+
+{{ #include @code/recipes/how-to-spawn-vim/src/main.rs:handle-events }}
+```
+
+In the `main` function, we initialize the terminal and enter a loop where we draw the UI and handle
+events. The `handle_events` function listens for key events and returns an `Action` based on the key
+pressed.
+
+## Spawning Vim
+
+Now, let's define the function `run_editor` function attached to `Action::EditFile` action.
+
+```rust title="main.rs"
+{{ #include @code/recipes/how-to-spawn-vim/src/main.rs:run_editor }}
+```
+
+To spawn Vim from our TUI app, we first need to relinquish control of input and output, allowing Vim
+to have full control over the terminal.
+
+The `run_editor` function handles the logic for spawning vim. First, we leave the alternate screen
+and disable raw mode to restore terminal to it's original state. This part is similar to what
+`restore_terminal` function does in the
+[full code](https://github.com/ratatui-org/ratatui-website/tree/main/code/how-to-spawn-vim/src/main.rs).
```

Comment:
mention the new ratatui::restore function here

Reply:
I updated this part to

```md
The `run_editor` function handles the logic for spawning vim. First, we leave the alternate screen
and disable raw mode to restore terminal to it's original state. This part is similar to what
[`ratatui::restore`](https://github.com/ratatui/ratatui/blob/b13e2f94733afccfe02275fca263bde1dc532d2f/src/terminal/init.rs#L206-L232)
function does in the `main` function. 
```

Should I include a permalink to `try_restore` function or not include a link at all?
https://github.com/ratatui/ratatui/blob/b13e2f94733afccfe02275fca263bde1dc532d2f/src/terminal/init.rs#L206-L232

Reply:
It's probably better to link to docs.rs instead of code.

Reply:
Is this link correct? https://docs.rs/ratatui/latest/ratatui/fn.try_restore.html

Reply:
I just force pushed with that link. I have also updated rendered page screenshot in PR description

Reply:
just restore rather than try_restore is the preferred link I think. https://docs.rs/ratatui/latest/ratatui/fn.restore.html

Reply:
I just fixed the link with a force push :+1: